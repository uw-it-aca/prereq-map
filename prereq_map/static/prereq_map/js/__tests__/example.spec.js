import { mount } from '@vue/test-utils'
import Onbaording from '../components/onboarding.vue'

const { toMatchImageSnapshot } = require('jest-image-snapshot');

//import { toMatchImageSnapshot } from 'jest-image-snapshot'
expect.extend({ toMatchImageSnapshot });

const puppeteer = require('puppeteer');

describe('Example Component', () => {

    let browser;

    beforeAll(async () => {
        browser = await puppeteer.launch({
          args: ['--disable-dev-shm-usage']
        });
    });

    test('is a Vue instance', () => {
        const wrapper = mount(Onbaording)
        expect(wrapper.isVueInstance()).toBeTruthy()
    });

    test('renders correctly', () => {
        const wrapper = mount(Onbaording)
        expect(wrapper.element).toMatchSnapshot()
    });

    test('visual regression', async () => {

        const page = await browser.newPage();
        await page.goto('http://localhost:8123/course-search/');
        const screenshot = await page.screenshot();
        expect(screenshot).toMatchImageSnapshot();


    });

    afterAll(async () => {
        await browser.close();
    });

})
