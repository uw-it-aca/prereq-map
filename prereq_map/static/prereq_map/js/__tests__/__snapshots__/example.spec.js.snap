// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example Component renders correctly 1`] = `
<div>
  <div
    aria-hidden="true"
    aria-labelledby=""
    class="modal fade"
    id="onboardingModal"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-lg modal-dialog-centered"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <button
            aria-label="Close"
            class="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
         
        <div
          class="modal-body"
        >
          <div
            class="carousel slide"
            data-interval="false"
            data-ride="carousel"
            id="carouselExampleIndicators"
          >
            <ol
              class="carousel-indicators"
            >
              <li
                class="active"
                data-slide-to="0"
                data-target="#carouselExampleIndicators"
              />
               
              <li
                data-slide-to="1"
                data-target="#carouselExampleIndicators"
              />
               
              <li
                data-slide-to="2"
                data-target="#carouselExampleIndicators"
              />
               
              <li
                data-slide-to="3"
                data-target="#carouselExampleIndicators"
              />
            </ol>
             
            <div
              class="carousel-inner rounded-lg"
            >
              <div
                class="carousel-item active"
                style="background: purple;"
              >
                <h2
                  class="text-white mb-5"
                >
                  Discover courses and plan your schedule more effectively
                </h2>
                 
                <p
                  class="text-white"
                >
                  It can be tough figuring out which courses to take. How do you find courses similar to ones you’ve really enjoyed? Of the courses you’ve already taken, which ones provide a foundation for more advanced coursework? What sequence of courses is best? The Prereq Map helps you discover interesting courses and enables you to be strategic about planning your course schedule. (What’s a prereq? Prereq is short for prerequisites: the courses that act as a foundation for other courses.)
                </p>
                 
                <p
                  class="text-white"
                >
                  The Prereq Map shows current prerequisites. When planning several quarters out, keep in mind that prerequisites may change. Also note that the Prereq Map is based solely on course prerequisites and does not take into account specific requirements for graduation.
                </p>
                 
                <div
                  class="text-center mt-5"
                >
                  <div>
                    <a
                      href="#"
                      target="_blank"
                    >
                      Further details
                    </a>
                  </div>
                </div>
                 
                <div
                  class="text-center mt-2"
                >
                  <button
                    class="btn btn-primary"
                  >
                    Got it!
                  </button>
                </div>
              </div>
               
              <div
                class="carousel-item"
              >
                <h2
                  class="mb-5"
                >
                  See the prerequisite map for a curriculum (e.g. Biology, BIOL)
                </h2>
                 
                <p>
                  Begin exploring the Prereq Map by choosing a curricula (subjects comprising a course of study) you’re interested in. Follow the lines to see the prerequisites for courses. View courses that have prereqs and courses that are prereqs for other courses. 
                </p>
              </div>
               
              <div
                class="carousel-item"
              >
                <h2
                  class="mb-5"
                >
                  Get additional info about a course
                </h2>
                 
                <p>
                  Each node on the map represents a specific course. Click the node to view prerequisite information for the course. Click the links inside the node to learn more.
                </p>
              </div>
               
              <div
                class="carousel-item"
              >
                <h2
                  class="mb-5"
                >
                  Get more course detail and find related curricula.
                </h2>
                 
                <p>
                  On the Course Search page, click links to browse related course offerings. You can also explore related curricula and their prerequisite maps.
                </p>
              </div>
            </div>
             
            <a
              class="carousel-control-prev"
              data-slide="prev"
              href="#carouselExampleIndicators"
              role="button"
            >
              <span
                aria-hidden="true"
                class="carousel-control-prev-icon"
              />
               
              <span
                class="sr-only"
              >
                Previous
              </span>
            </a>
             
            <a
              class="carousel-control-next"
              data-slide="next"
              href="#carouselExampleIndicators"
              role="button"
            >
              <span
                aria-hidden="true"
                class="carousel-control-next-icon"
              />
               
              <span
                class="sr-only"
              >
                Next
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
